demos/imagebuilder-windowsbaseline/main.bicep

```diff
@@ -0,0 +1,212 @@
+          'Microsoft.Resources/deploymentScripts/write'
+          'Microsoft.VirtualMachineImages/imageTemplates/run/action'
+        ]
+      }
+    ]
+    assignableScopes: [
+      resourceGroup().id
+    ]
+  }
+}
+
+resource roleAssignmentName_resource 'Microsoft.Authorization/roleAssignments@2021-04-01-preview' = {
+  name: roleAssignmentName
+  properties: {
+    roleDefinitionId: roleDefinitionName_resource.id
+    principalId: reference(identityName_resource.id, '2018-11-30').principalId
+    scope: resourceGroup().id
+    principalType: 'ServicePrincipal'
+  }
+}
+
+resource imageGalleryName_resource 'Microsoft.Compute/galleries@2021-07-01' = {
+  name: imageGalleryName
+  location: location
+  properties: {}
+}
+
+resource imageGalleryName_imageDefinition_imageDefinitionName 'Microsoft.Compute/galleries/images@2020-09-30' = {
+  parent: imageGalleryName_resource
+  name: '${imageDefinition.imageDefinitionName}'
+  location: location
+  properties: {
+    osType: 'Windows'
+    osState: 'Generalized'
+    identifier: {
+      publisher: imageDefinition.publisher
+      offer: imageDefinition.offer
+      sku: imageDefinition.sku
+    }
+    recommended: {
+      vCPUs: {
+        min: 2
+        max: 8
+      }
+      memory: {
+        min: 16
+        max: 48
+      }
+    }
+    hyperVGeneration: 'V1'
+  }
+}
+
+resource imageTemplateName_resource 'Microsoft.VirtualMachineImages/imageTemplates@2020-02-14' = {
+  name: imageTemplateName
+  location: location
+  identity: {
+    type: 'UserAssigned'
+    userAssignedIdentities: {
+      '${identityName_resource.id}': {}
+    }
+  }
+  properties: {
+    buildTimeoutInMinutes: 60
+    vmProfile: {
+      vmSize: 'Standard_D1_v2'
+      osDiskSizeGB: 127
+    }
+    source: {
+      type: 'PlatformImage'
+      publisher: 'MicrosoftWindowsServer'
+      offer: 'WindowsServer'
+      sku: '2019-Datacenter'
+      version: 'latest'
+    }
+    customize: [
+      {
+        type: 'WindowsUpdate'
+        searchCriteria: 'IsInstalled=0'
+        filters: [
+          'exclude:$_.Title -like \'*Preview*\''
+          'include:$true'
+        ]
+        updateLimit: 40
+      }
+      {
+        type: 'PowerShell'
+        name: 'AzureWindowsBaseline'
+        runElevated: true
+        scriptUri: customizerScriptUri
+      }
+    ]
+    distribute: [
+      {
+        type: 'SharedImage'
+        galleryImageId: imageGalleryName_imageDefinition_imageDefinitionName.id
+        runOutputName: runOutputName
+        replicationRegions: replicationRegions
+      }
+    ]
+  }
+  dependsOn: [
+    imageGalleryName_resource
+  ]
+}
+
+resource Image_template_build 'Microsoft.Resources/deploymentScripts@2020-10-01' = {
+  name: 'Image_template_build'
+  location: location
+  kind: 'AzurePowerShell'
+  identity: {
+    type: 'UserAssigned'
+    userAssignedIdentities: {
+      '${identityName_resource.id}': {}
+    }
+  }
+  properties: {
+    forceUpdateTag: 'newGuid()'
```

Comment:
this isn't processed as an expresssion, you have it in quotes so it's just a string...  If you want to force this to run every time, you have to use newGuid() in a parameter's defaultValue

Reply:
ack